meveo-admin/ejbs/src/main/java/org/meveo/service/base/BusinessService.java

```diff
@@ -169,6 +179,87 @@ public BusinessEntity findByEntityClassAndCode(Class<?> clazz, String code) {
         }
     }
 
+    /** 
+     * Find which the module the entity belongs to 
+
+     * @param entity
+     * @return MeveoModule, or null otherwise
+     */
+    @SuppressWarnings({ "rawtypes", "deprecation" })
+	public MeveoModule findModuleOf(P entity){
+    	MeveoModule module = null;
+    	if (entity != null) {
+    		Session session = this.getEntityManager().unwrap(Session.class);
+    		Query q = session.createQuery("SELECT m.module FROM ModuleItem m WHERE m.code = :code AND m.ItemType = :itemType");
```

Comment:
m.ItemType => m.itemType : the case always count

Reply:
You should probably take care of the "applies_to" field when it's provided (check if entity is instance of CustomFieldTemplate)